flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphTestUtils.java

```diff
@@ -428,6 +428,18 @@ public static Execution getExecution(
         return ejv.getTaskVertices()[subtaskIndex].getCurrentExecutionAttempt();
     }
 
+    public static ExecutionAttemptID createExecutionAttemptId() {
+        return createExecutionAttemptId(new JobVertexID(0, 0), 0, 0);
+    }
+
+    public static ExecutionAttemptID createExecutionAttemptId(
+            JobVertexID jobVertexId, int subtaskIndex, int attemptNumber) {
+        return new ExecutionAttemptID(
+                new ExecutionGraphID(),
+                new ExecutionVertexID(jobVertexId, subtaskIndex),
+                attemptNumber);
+    }
```

Comment:
I'd rather have these factory methods in the ExecutionAttemptID class so that they are available everywhere.

Reply:
I prefer to keep them as test code to avoid messing up production code for testing purposes. Some other concerns see my other [comment](https://github.com/apache/flink/pull/19747#discussion_r875516319).